src/components/DataTable/TableHeader.js

```diff
@@ -12,11 +12,18 @@ const translationKeys = {
 const translateWithId = (key, { sortDirection, isSortHeader, sortStates }) => {
   if (key === translationKeys.iconDescription) {
     if (isSortHeader) {
-      const order =
-        sortDirection === sortStates.DESC ? 'descending' : 'ascending';
-      return `Sort rows by this header in ${order} order`;
+      // When transitioning, we know that the sequence of states is as follows:
+      // NONE -> ASC -> DESC -> NONE
+      if (sortDirection === sortStates.NONE) {
+        return 'Rort rows by this header in ascending order';
```

Comment:
@joshblack Typo `Rort` vs `Sort`?
```suggestion
        return 'Sort rows by this header in ascending order';
```